File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        for i in range(len(nums)):
            rest = target - nums[i]
            rest_nums = nums[i+1:]
            if rest in rest_nums:
                result.append(i)
                result.append(rest_nums.index(rest)+i+1)
                break
        return result

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let indices = new Map();

    for (let i = 0; i < nums.length; i++) {
        diff = target - nums[i];
        if (indices.has(diff)) {
            return [i, indices.get(diff)];
        }
        indices.set(nums[i], i);
    }
};